replica/distributed_loader.cc

```diff
@@ -460,25 +460,20 @@ future<> table_populator::populate_subdir(sstables::sstable_state state, allow_o
 }
 
 future<> distributed_loader::populate_keyspace(distributed<replica::database>& db, keyspace& ks, sstring ks_name) {
-    // might have more than one dir for a keyspace iff data_file_directories is > 1 and
-    // somehow someone placed sstables in more than one of them for a given ks. (import?)
-    return parallel_for_each(db.local().get_config().data_file_directories(), [&db, &ks, ks_name] (const sstring& data_dir) {
-        return populate_keyspace(db, ks, data_dir, ks_name);
-    });
-}
-
-future<> distributed_loader::populate_keyspace(distributed<replica::database>& db, keyspace& ks, sstring datadir, sstring ks_name) {
-    auto ksdir = datadir + "/" + ks_name;
-
     dblog.info("Populating Keyspace {}", ks_name);
     auto& tables_metadata = db.local().get_tables_metadata();
 
     co_await coroutine::parallel_for_each(ks.metadata()->cf_meta_data() | boost::adaptors::map_values, [&] (schema_ptr s) -> future<> {
         auto uuid = s->id();
+        sstring cfname = s->cf_name();
         lw_shared_ptr<replica::column_family> cf = tables_metadata.get_table(uuid).shared_from_this();
 
-        sstring cfname = cf->schema()->cf_name();
-        dblog.info("Keyspace {}: Reading CF {} id={} version={} storage={}", ks_name, cfname, uuid, s->version(), cf->get_storage_options().type_string());
+      // might have more than one dir for a keyspace iff data_file_directories is > 1 and
+      // somehow someone placed sstables in more than one of them for a given ks. (import?)
+      co_await coroutine::parallel_for_each(db.local().get_config().data_file_directories(), [&] (const sstring& datadir) -> future<> {
```

Comment:
Why don't you want to get_table_on_all_shards() first and then walk its `_config.all_datadirs` instead?

Reply:
Good idea.  Will do.